projects/tour-of-heroes-data/src/app/villains/villain-detail/villain-detail.component.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Villain } from '../../core';
+
+@Component({
+  selector: 'ngrx-villain-detail',
+  templateUrl: './villain-detail.component.html',
+  styleUrls: ['./villain-detail.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VillainDetailComponent implements OnChanges {
+  @Input() villain: Villain;
```

Comment:
```suggestion
  @Input() villain!: Villain | Record<string, never>;
```

`Record<string, never>` because villain will be empty object in create mode.